apollo-router/src/spec/fragments.rs

```diff
@@ -15,10 +19,45 @@ use crate::spec::SpecError;
 
 #[derive(Debug, Default, Serialize, Deserialize)]
 pub(crate) struct Fragments {
-    map: HashMap<String, Fragment>,
+    pub(crate) map: HashMap<String, Fragment>,
 }
 
 impl Fragments {
+    pub(crate) fn from_hir(compiler: &ApolloCompiler, schema: &Schema) -> Result<Self, SpecError> {
+        let map = compiler
```

Comment:
the code is so much nicer. 30 lines VS 90 lines when doing it manually with the parser